language/use-cases/sql-code-generation/sql_code_generation_langchain.ipynb

```diff
@@ -0,0 +1,1186 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #86.    def generate_prompt(user_prompt):
</pre><p>I suggest breaking out this helper function definition (and all its dependencies like MaxMarginalRelevanceExampleSelector) into their own section right after "Build schema definition" but before "Generate SQL queries" (and associated helper functions generate/execute sql).</p><p>Would be more clear for end-user if they build the prompt fully (LLM input) before being presented with code to generate / execute queries (LLM output).</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/404/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Reply:
<p>generate_prompt. -&gt; build_prompt</p>

Reply:
<p>Done</p>